src/node.ts

```diff
@@ -284,6 +307,22 @@ class Node extends VEventTarget {
   log = (...args: unknown[]): void => {
     this.dispatchEvent(new VEvent('log', { detail: { args } }));
   };
+
+  /**
+   * Method that sets resetState to false
+   */
+  keepState = (): void => {
+    this.resetState = false;
+  };
+
+  /**
+   * Checks if the node has state.
+   * @returns True if node has state, false if doesn't.
+   */
+  hasState(): boolean {
+    if (!this.resolved) throw new Error('Node is not resolved');
```

Comment:
Since we punish those who call `hasState()` without first checking if the node is resolved, we have to provide also `isResolved()` method, and declare `this.resolved` as private.